Extend/Model/Order.php

```diff
@@ -246,4 +248,19 @@ public function getTranslatedState()
     {
         return self::getTranslatedStates()[$this->oxorder__wdoxidee_orderstate->value] ?? '';
     }
+
+    /**
+     *
+     * Create a temporary Order that should not be saved but used for creating a
+     * {@link Wirecard\PaymentSdk\Transaction\Transaction}
+     *
+     * @param Basket $oBasket
+     * @param User   $oUser
+     */
+    public function createTemp($oBasket, $oUser)
+    {
+        $this->_setUser($oUser);
+        $this->_loadFromBasket($oBasket);
+        $this->oxorder__oxid = new Field(Registry::getSession()->getVariable('sess_challenge'));
```

Comment:
The session challenge is the oxid order id?

Reply:
yes, see `OxidEsales\Eshop\Application\ModelOrder#finalizeOrder`, that is the class we extend, and there they use the `sess_challenge` for the order id.